software/examples/ensogen.cpp

```diff
@@ -101,19 +129,239 @@
 // Number of transfers required to send a buffer full of packets.
 #define TRANSFERS_PER_BUFFER (((BUFFER_SIZE - 1) / enso::kMaxTransferLen) + 1)
 
+// Macros for cmd line option names
+#define CMD_OPT_HELP "help"
+#define CMD_OPT_COUNT "count"
+#define CMD_OPT_CORE "core"
+#define CMD_OPT_QUEUES "queues"
+#define CMD_OPT_SAVE "save"
+#define CMD_OPT_SINGLE_CORE "single-core"
+#define CMD_OPT_RTT "rtt"
+#define CMD_OPT_RTT_HIST "rtt-hist"
+#define CMD_OPT_RTT_HIST_OFF "rtt-hist-offset"
+#define CMD_OPT_RTT_HIST_LEN "rtt-hist-len"
+#define CMD_OPT_STATS_DELAY "stats-delay"
+#define CMD_OPT_PCIE_ADDR "pcie-addr"
+
 static volatile int keep_running = 1;
 static volatile int force_stop = 0;
 static volatile int rx_ready = 0;
 static volatile int rx_done = 0;
 static volatile int tx_done = 0;
 
+using enso::Device;
+using enso::RxPipe;
+using enso::TxPipe;
+
+/******************************************************************************
+ * Structure Definitions
+ *****************************************************************************/
+/*
```

Comment:
I've updated all the function headers. Leaving banners like these since they are meant for reading the source code file. I'm trying to render this file in doxygen and see if my changes are proper.  I tried running `doxygen 
Doxyfile.in` and I got some errors [doxygen.txt](https://github.com/crossroadsfpga/enso/files/15080443/doxygen.txt). Can you refer me to some link to run this and verify? Thanks.